### PR TITLE
feat(email): When primary email bouncy, send to secondary email

### LINKED ISSUE #1953: If your primary email is bouncy, we don't send to your secondary email
I could have this wrong here, but my read of the code suggests that our sending logic goes like this:

* in `lib/routes/account.js`, we call e.g. `sendVerifyLoginEmail` using the primary address as the "To:" address, and the secondary addresses as "CC:" addresses.
* in `lib/senders/index.js`, we check whether the "To:" address has a history of bounces, and if so then we error out.

So if your primary email address is bouncy, we'll just refuse to send the signin confirmation email at all, and adding a secondary email won't help you.  @seanmonstar @vbudhram thoughts?

### PR DESCRIPTION
**Pull request by vbudhram**
This PR does a couple things to verify we are not sending to bouncy emails.

* Don't send to any bounced secondary emails
* If primary email bounced and their exisits a secondary email, send to that email.

Fixes #1953

### CODE REVIEW

File: lib/routes/account.js [hunk 1]
```diff
@@ -929,7 +929,11 @@ module.exports = (
             && ! doSigninConfirmation
             && emailRecord.emailVerified
           if (shouldSendNewDeviceLoginEmail) {
-            const secondaryEmails = features.isSecondaryEmailEnabled(emailRecord.email) ? db.accountEmails(sessionToken.uid) : P.resolve([])
+            const secondaryEmails = features.isSecondaryEmailEnabled(emailRecord.email) ? db.accountEmails(sessionToken.uid) : P.resolve([{
```
Comment (vbudhram) [inline lib/routes/account.js]: I am not sure if we need `features.isSecondaryEmailEnabled` check here. All mailer functions accept an array of emails as first parameter.

**Comment by vbudhram** [comment]
@mozilla/fxa-devs r?

**Comment by vbudhram** [comment]
Moving back to WIP, can make the code more concise.

**Comment by vbudhram** [comment]
Cleaned up the code a bit, this should be ready now.

@mozilla/fxa-devs r?

File: lib/senders/index.js [hunk 2]
```diff
@@ -52,6 +90,18 @@ module.exports = function (log, config, error, bounces, translator, sender) {
...
+    function getPrimaryEmail(emails) {
+      var primaryEmail
+      for (var i=0; i<emails.length; i++) {
+        if (emails[i].isPrimary) {
+          primaryEmail = emails[i]
+          break
+        }
+      }
+
+      return primaryEmail ? primaryEmail.email : primaryEmail
```
Comment (seanmonstar) [inline lib/senders/index.js]: `return primaryEmail && primaryEmail.email`

**Comment by vbudhram** [comment]
Updated per @seanmonstar, @mozilla/fxa-devs final r?

**Pull request merged**